File: backend/src/lib/lemma/docstore.py
from .datastore import get_profile, save_profile, get_applications, update_application

def get_resume(user_email: str):
    profile = get_profile(user_email)
    return profile.get("resume", "") if profile else ""

def save_resume(user_email: str, resume_text):
    profile = get_profile(user_email) or {}
    profile["resume"] = resume_text
    profile["resumeText"] = resume_text
    return save_profile(user_email, profile) is not None

def get_tailored_resume(user_email: str, app_id):
    apps = get_applications(user_email)
    for app in apps:
        if app.get("id") == app_id:
            return app.get("tailoredBullets", "")
    return ""

def save_tailored_resume(user_email: str, app_id, tailored_bullets):
    return update_application(user_email, app_id, {"tailoredBullets": tailored_bullets})

File: frontend/src/lib/lemma/docstore.js
import fs from 'fs';
import path from 'path';

const DB_PATH = path.join(process.cwd(), 'src', 'lib', 'lemma', 'db.json');

function initDb() {
  const dir = path.dirname(DB_PATH);
  if (!fs.existsSync(dir)) {
    fs.mkdirSync(dir, { recursive: true });
  }
  if (!fs.existsSync(DB_PATH)) {
    fs.writeFileSync(DB_PATH, JSON.stringify({ applications: [], profile: null }, null, 2));
  }
}

export function getResume() {
  initDb();
  try {
    const data = JSON.parse(fs.readFileSync(DB_PATH, 'utf-8'));
    return data.profile?.resume || '';
  } catch (error) {
    console.error('Error fetching resume from Lemma DocStore:', error);
    return '';
  }
}

export function saveResume(resumeText) {
  initDb();
  try {
    const data = JSON.parse(fs.readFileSync(DB_PATH, 'utf-8'));
    if (!data.profile) {
      data.profile = {};
    }
    data.profile.resume = resumeText;
    fs.writeFileSync(DB_PATH, JSON.stringify(data, null, 2));
    return true;
  } catch (error) {
    console.error('Error saving resume to Lemma DocStore:', error);
    return false;
  }
}

export function getTailoredResume(appId) {
  initDb();
  try {
    const data = JSON.parse(fs.readFileSync(DB_PATH, 'utf-8'));
    const applications = data.applications || [];
    const app = applications.find(a => a.id === appId);
    return app ? app.tailoredBullets : '';
  } catch (error) {
    console.error('Error fetching tailored resume from Lemma DocStore:', error);
    return '';
  }
}

export function saveTailoredResume(appId, tailoredBullets) {
  initDb();
  try {
    const data = JSON.parse(fs.readFileSync(DB_PATH, 'utf-8'));
    const applications = data.applications || [];
    const index = applications.findIndex(a => a.id === appId);
    if (index >= 0) {
      applications[index].tailoredBullets = tailoredBullets;
      data.applications = applications;
      fs.writeFileSync(DB_PATH, JSON.stringify(data, null, 2));
      return true;
    }
    return false;
  } catch (error) {
    console.error('Error saving tailored resume to Lemma DocStore:', error);
    return false;
  }
}
